PYME/recipes/kinetics_recipes.py

```diff
@@ -0,0 +1,428 @@
+from PYME.recipes.base import register_module, ModuleBase, Filter, OutputModule
+from PYME.recipes.traits import Input, Output, Float, Enum, CStr, Bool, Int, List, DictStrStr, DictStrList, Either, \
+    ListFloat, ListStr, Button, ToolbarButton, Array, Directory, File
+import numpy as np
+import pylab
+from PYME.IO import tabular, MetaDataHandler
+from PYME.Analysis import MetaData
+from PYME.LMVis.pipeline import *
+from PYME.Analysis._fithelpers import FitModelWeighted, FitModel
+from PYME.Analysis.points.DeClump.pyDeClump import mergeClumpsDyeKinetics
+
+
+colours = ['r', 'k', 'b', 'g']
+
+@register_module('DyeKineticsInit')
+class DyeKineticsInit(ModuleBase):
+    """
+    Module which sets up everything for the actual dye kinetics code.
+    A workaround for the weird way recipes are run when modules are added together without an explicitly defined UI,
+    as in the bakeshop
+
+    Without this, the only UI we see is that of the first module, which is the clump finder in this case. While this isn't
+    100% needed to run the script, user-customizability would be nil, making it harder to use.
+    """
```

Comment:
Can you help me understand what you mean by this? A null execute method is a huge red flag. Also if I recall you've been developing with a fairly outdated version of PYME. David's put a lot into fixing up the pipeline (left side of VisGUI / PYMEVisualize) recipe view / linking that more firmly with the recipe tab in VisGUI. Maybe have a check that this still does what you think/want it to?

Reply:
You're right - this is definitely not the right way to call a recipe. I had in it my head this should be called from the dropdown menus, and it still does what I want it to, but with the whole analysis pipeline stored as a recipe it won't be necessary any more